CSharpMath/Structures/Dictionary.cs

```diff
@@ -0,0 +1,278 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace CSharpMath.Structures {
+
+  [SuppressMessage("Naming", "CA1710:Identifiers should have correct suffix", Justification = NotACollection)]
+  [SuppressMessage("Design", "CA1010:Collections should implement generic interface", Justification = NotACollection)]
+  public class ProxyAdder<TKey, TValue> : IEnumerable {
+    const string NotACollection = "This is not a collection. It implements IEnumerable just to support collection initializers.";
+    [Obsolete(NotACollection, true)]
+    [SuppressMessage("Design", "CA1033:Interface methods should be callable by child types", Justification = NotACollection)]
+    IEnumerator IEnumerable.GetEnumerator() => throw new NotSupportedException(NotACollection);
+    public ProxyAdder(Action<TKey, TValue>? added = null) => Added += added;
+    public event Action<TKey, TValue>? Added;
+    public void Add(TKey key1, TValue value) => Added?.Invoke(key1, value);
```

Comment:
This is the only one that is needed.

Reply:
No. Other `Add` methods are needed as well. Not all of them but I suppose it is fine to have more methods just in case.

VS not always counts references correctly.

Reply:
Agreed with @FoggyFinder. Collection initializers bind to `Add` directly, and without the overloads, e.g. ` { "a", "b", new Ordinary("c") } `will not work.